packages/docusaurus-plugin-client-redirects/src/normalizePluginOptions.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {PluginOptions, RedirectsCreator, UserPluginOptions} from './types';
+import * as Yup from 'yup';
+
+export const DefaultPluginOptions: PluginOptions = {
+  fromExtensions: [],
+  toExtensions: [],
+};
+
+function isRedirectsCreator(value: any): value is RedirectsCreator | undefined {
+  if (value === null || typeof value === 'undefined') {
+    return true;
+  }
+  return value instanceof Function;
+}
+
+const UserOptionsSchema = Yup.object().shape<UserPluginOptions>({
```

Comment:
I've included Yup because I think it's a good solution to validate unsafe user inputs like plugin options. It gives a way to show a better error to the user instead of failing late with a cryptic runtime error